pkg/checks/traceroute/traceroute.go

```diff
@@ -58,15 +66,46 @@ func tcpHop(ctx context.Context, addr net.Addr, ttl int, timeout time.Duration)
 			},
 		}
 
+		span.AddEvent("Attempting TCP connection", trace.WithAttributes(
+			attribute.String("remote_addr", addr.String()),
+			attribute.Int("ttl", ttl),
+			attribute.Int("port", port),
+		))
+
 		// Attempt to connect to the target host
 		conn, err := dialer.DialContext(ctx, "tcp", addr.String())
-		if !errors.Is(err, unix.EADDRINUSE) {
+
+		switch {
+		case err == nil:
+			span.AddEvent("TCP connection succeeded", trace.WithAttributes(
+				attribute.Stringer("remote_addr", addr),
+				attribute.Int("ttl", ttl),
+				attribute.Int("port", port),
+			))
+			return conn, port, nil
+		case errors.Is(err, unix.EADDRINUSE):
+			// Address in use, retry by continuing the loop
+			continue
+		case errors.Is(err, syscall.EHOSTUNREACH):
+			// No route to host is no error because of how tcp traceroute works
+			// we are expecting the connection to fail because of TTL expiry
+			span.SetStatus(codes.Unset, "No route to host")
+			span.AddEvent("No route to host", trace.WithAttributes(
+				attribute.String("error", err.Error()),
+			))
+			logger.FromContext(ctx).DebugContext(ctx, "No route to host", "error", err.Error())
 			return conn, port, err
 		}
+
+		span.AddEvent("TCP connection failed", trace.WithAttributes(
```

Comment:
Can we move this to the default case, as we're already doing the error handling there?